spectral_cube/lower_dimensional_structures.py

```diff
@@ -1093,6 +1120,51 @@ def __getattribute__(self, attrname):
 
 class VaryingResolutionOneDSpectrum(OneDSpectrum, MultiBeamMixinClass):
     def __new__(cls, value, beams=None, **kwargs):
+        out = super(VaryingResolutionOneDSpectrum, cls).__new__(cls, value, **kwargs)
         if beams is not None:
-            cls.beams = beams
-        return super(VaryingResolutionOneDSpectrum, cls).__new__(cls, value, **kwargs)
+            out.beams = beams
```

Comment:
Needed this to trigger the `beams.setter` checks below. Setting `cls.beams`, then running `super` didn't do this.